slowapi/middleware.py

```diff
@@ -1,56 +1,152 @@
+from typing import Optional, Iterable, Tuple
+
 from starlette.applications import Starlette
+from starlette.datastructures import MutableHeaders
 from starlette.middleware.base import (
     BaseHTTPMiddleware,
     RequestResponseEndpoint,
 )
 from starlette.requests import Request
 from starlette.responses import Response
-from starlette.routing import Match
+from starlette.routing import BaseRoute, Match
+from starlette.types import ASGIApp, Message, Scope, Receive, Send
 
 from slowapi import Limiter, _rate_limit_exceeded_handler
 
 
+def _find_route_handler(routes: Iterable[BaseRoute], scope: Scope):
+    handler = None
+    for route in routes:
+        match, _ = route.matches(scope)
+        if match == Match.FULL and hasattr(route, "endpoint"):
+            handler = route.endpoint  # type: ignore
+    return handler
+
+
+def _get_route_name(handler):
```

Comment:
```suggestion
def _get_route_name(handler: Callable) -> str:
```

Reply:
this would raise some linting errors `cannot find __module__ in (...) -> Any`, so I used Type[Callable] instead.